vimeo/VimeoNetworking
# VimeoNetworking clean up

#### Ticket

[VIM-7216](https://vimean.atlassian.net/browse/VIM-7216)

#### Pull Request Checklist


#### Issue Summary

While doing some prep research on VimeoNetworking I came across old/unused code that could be removed. The purpose of this ticket is to account for the clean up work.

Also see [related PR](https://github.vimeows.com/MobileApps/VimeoKit/pull/468/) on VimeoKit

#### Implementation Summary

- Most of the changes are for removing unused code
- Added nullability annotations to VIMModelObject
- Refactored Mappable conformance to avoid duplicated implementation of validateModel()

#### How to Test

- Build and run, make sure all unit tests pass
- Update the main Vimeo app to point to this branch on both `VimeoNetworking` and `VimeoKit`, build and run and ensure all tests pass
